src/EppoClient.php

```diff
@@ -117,9 +123,18 @@ public static function getInstance(): EppoClient
      * @throws InvalidArgumentException
      * @throws SimpleCacheInvalidArgumentException
      */
-    public function getStringAssignment(string $subjectKey, string $flagKey, array $subjectAttributes = []): ?string {
-        $assignmentVariation = $this->getAssignmentVariation($subjectKey, $flagKey, $subjectAttributes, self::VARIANT_TYPE_STRING);
-        return  $assignmentVariation ? strval($assignmentVariation->typedValue) : null;
+    public function getStringAssignment(string $subjectKey, string $flagKey, array $subjectAttributes = []): ?string
+    {
+        try {
+            $assignmentVariation = $this->getAssignmentVariation($subjectKey, $flagKey, $subjectAttributes, self::VARIANT_TYPE_STRING);
+            return  $assignmentVariation ? strval($assignmentVariation->typedValue) : null;
+        } catch (Exception $exception) {
+            if ($this->isGracefulMode) {
+                error_log('[Eppo SDK] Error getting string assignment: ' . $exception->getMessage());
+                return null;
+            }
+            throw $exception;
```

Comment:
Like we did in the JS client, this code can be deduped into a private helper function